pkg/util/tls.go

```diff
@@ -75,13 +126,29 @@ func NewTLSConfigFromClientConfiguration(configuration *configuration.ClientConf
 	}
 	tlsConfig.ServerName = configuration.ServerName
 
-	if configuration.ClientCertificate != "" && configuration.ClientPrivateKey != "" {
-		// Serve a client certificate when provided.
-		cert, err := tls.X509KeyPair([]byte(configuration.ClientCertificate), []byte(configuration.ClientPrivateKey))
-		if err != nil {
-			return nil, StatusWrapWithCode(err, codes.InvalidArgument, "Invalid client certificate or private key")
+	// No client TLS is used when this is unset.
+	if configuration.ClientKeyPair != nil {
+		// Require the use of server-side certificates.
```

Comment:
Could we factor out this code, so that client/server creation can use the same code?